homework/python_processing.py

```diff
@@ -0,0 +1,174 @@
+def prim_key_checker(df):
+    """
+    Assumes that primary key is the first column of the dataframe
+    Check whether primary key variable has null values or whether it is unique
+    """
+    if df.iloc[:,0].isnull().values.any() | df.iloc[:,0].nunique()<len(df.iloc[:,0]):
+        raise Exception(f'There is a problem with the primary key!!')
+    else:
+        print("Primary key is in good format")
+
+def new_table_inspector(df):
+    print("The shape of the dataframe is: {}".format(df.shape))
+    print("The head of the dataframe is:")
+    print(df.head())
+    print(prim_key_checker(df))
+
+
+
+# I. EMPLOYEE TABLE
+# Split data into new, normalized tables according to 3NF
+employee=employees[['employee_number','last_name','first_name','reports_to',
+                    'job_title','office_code']].drop_duplicates()
+
+
+# Inspect datasets
+print(new_table_inspector(employee))
+
+
+# II. OFFICE TABLE
+# Split data into new, normalized tables according to 3NF
+office=employees[['office_code','city','state','country',
+                    'office_location',]].drop_duplicates()
+
+# Inspect datasets
+print(new_table_inspector(office))
+
+# III. PRODUCTS TABLE
+# Split data into new, normalized tables according to 3NF
+products=products[['product_code','product_name','product_line','product_scale',
+                     'product_vendor','product_description',
+                    'quantity_in_stock','buy_price',
+                    '_m_s_r_p','html_description']].drop_duplicates()
+
+# Inspect datasets
+print(new_table_inspector(products))
+
+
+# IV. CUSTOMERS TABLE
+# Split data into new, normalized tables according to 3NF
+customers=orders[['customer_number','customer_name','contact_last_name','contact_first_name',
+                     'city','state',
+                    'country','sales_rep_employee_number',
+                    'credit_limit','customer_location']].drop_duplicates()
+
+# Inspect datasets
+print(new_table_inspector(customers))
+
+
+# V. ORDERS TABLE
+# Split data into new, normalized tables according to 3NF
+order=orders[['order_number','product_code','customer_number','quantity_ordered',
+                     'price_each','order_line_number',
+                    'order_date','required_date',
+                    'shipped_date','status','comments']].drop_duplicates()
+
+
+# Set a unique ID in the orders table for inspection purposes
+order.set_index(['order_number', 'product_code'])
+order["unique_id"]=order.index.values
+
+# Reset column order in the order table
+order = order[['unique_id','order_number','product_code','customer_number','quantity_ordered',
+                     'price_each','order_line_number',
+                    'order_date','required_date',
+                    'shipped_date','status','comments']]
+
+# Inspect datasets
+print(new_table_inspector(order))
+
+# Save modified dataframes as .csv files to import them directly from csv to PostgreSQL using Python
```

Comment:
I'm not sure I see the point of this. Speed? Copy from csv can be very fast, but psycopg2 offers execute_many transactions for speed, or you could copy from a in-memory file object without writing it first to disk (which is slow and unnecessary).